baseclasses/chemical_energy/uvvismeasurementconcentration.py

```diff
@@ -0,0 +1,218 @@
+        a_eln=dict(component='ReferenceEditQuantity', label='Concentration Detection'))
+
+    def normalize(self, archive, logger):
+        super(UVvisDataConcentration, self).normalize(archive, logger)
+
+        if self.chemical_composition_or_formulas:
+            if not archive.results:
+                archive.results = Results()
+            # if not archive.results.material:
+            archive.results.material = Material()
+            try:
+                from nomad.atomutils import Formula
+                formula = Formula(self.chemical_composition_or_formulas)
+                formula.populate(section=archive.results.material)
+            except Exception as e:
+                logger.warn('Could not analyse material', exc_info=e)
+
+        if self.chemical_composition_or_formulas == 'NH3':
+            if not self.estimated_peak_center:
+                self.estimated_peak_center = 650
+
+        if not self.peak_search_range:
+            self.peak_search_range = 20
+
+        if self.intensity is not None and self.wavelength is not None:
+            if self.chemical_composition_or_formulas is not None and self.peak_value is None:
+                if self.estimated_peak_center is not None and self.peak_search_range is not None:
+                    search_area_start = (self.estimated_peak_center - self.peak_search_range)
+                    search_area_end = (self.estimated_peak_center + self.peak_search_range)
+                else:
+                    search_area_start = min(self.wavelength)
+                    search_area_end = max(self.wavelength)
+                peak_area_df = pd.DataFrame({'intensity': self.intensity, 'wavelength': self.wavelength})
+                peak_area_df = peak_area_df.loc[(peak_area_df['wavelength'] >= search_area_start.magnitude) & (
+                    peak_area_df['wavelength'] <= search_area_end.magnitude)]
+                # since the search area can be set manually we assume that the peak is always the highest value in that range
+                peak_index = np.argmax(peak_area_df['intensity'])
+                peak = peak_area_df.iloc[peak_index]
+
+                self.peak_value = peak['intensity']
+                self.peak_wavelength = peak['wavelength']
+
+            fig = go.Figure(
+                data=[go.Scatter(name='UVvis', x=self.wavelength, y=self.intensity, mode='lines')])
+            fig.update_layout(xaxis_title=f'Wavelength [{self.wavelength.units}]',
+                              yaxis_title='Intensity',
+                              title_text='UVvis')
+            fig.update_layout(xaxis={'fixedrange': False})
+            if self.peak_value is not None and self.peak_wavelength is not None:
+                fig.add_traces(go.Scatter(x=[self.peak_wavelength.magnitude], y=[self.peak_value], mode='markers'))
+            self.figures = [PlotlyFigure(label='figure 1', figure=fig.to_plotly_json())]
+
+        if self.concentration is None:
```

Comment:
this should be done if no chemical composition is there if one changes the values this should be updated :)